src/control/server/mgmt_drpc_checker.go

```diff
@@ -0,0 +1,146 @@
+//
+// (C) Copyright 2022 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package server
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"google.golang.org/protobuf/proto"
+
+	srvpb "github.com/daos-stack/daos/src/control/common/proto/srv"
+	"github.com/daos-stack/daos/src/control/drpc"
+	"github.com/daos-stack/daos/src/control/system"
+)
+
+func (mod *srvModule) handleCheckerListPools(_ context.Context, reqb []byte) (out []byte, outErr error) {
+	// TODO: Remove if we never add request fields?
+	req := new(srvpb.CheckListPoolReq)
+	if err := proto.Unmarshal(reqb, req); err != nil {
+		return nil, drpc.UnmarshalingPayloadFailure()
+	}
+	mod.log.Debugf("handling CheckerListPools: %+v", req)
+
+	resp := new(srvpb.CheckListPoolResp)
+	defer func() {
+		mod.log.Debugf("CheckerListPools resp: %+v", resp)
+		out, outErr = proto.Marshal(resp)
+	}()
+
+	pools, err := mod.poolDB.PoolServiceList(true)
+	if err != nil {
+		mod.log.Errorf("failed to list pools: %s", err)
+		resp.Status = int32(drpc.DaosMiscError)
+		return
+	}
+
+	for _, ps := range pools {
+		resp.Pools = append(resp.Pools, &srvpb.CheckListPoolResp_OnePool{
+			Uuid:    ps.PoolUUID.String(),
+			Label:   ps.PoolLabel,
+			Svcreps: system.RanksToUint32(ps.Replicas),
+		})
+	}
+
+	return
+}
+
+func (mod *srvModule) handleCheckerRegisterPool(_ context.Context, reqb []byte) (out []byte, outErr error) {
+	req := new(srvpb.CheckRegPoolReq)
+	if err := proto.Unmarshal(reqb, req); err != nil {
+		return nil, drpc.UnmarshalingPayloadFailure()
+	}
+	mod.log.Debugf("handling CheckerRegisterPool: %+v", req)
+
+	resp := new(srvpb.CheckRegPoolResp)
+	defer func() {
+		mod.log.Debugf("CheckerRegisterPool resp: %+v", resp)
+		out, outErr = proto.Marshal(resp)
+	}()
+
+	uuid, err := uuid.Parse(req.Uuid)
+	if err != nil {
+		mod.log.Errorf("invalid pool UUID %q: %s", req.Uuid, err)
+		resp.Status = int32(drpc.DaosInvalidInput)
+		return
+	}
+	if !drpc.LabelIsValid(req.Label) {
+		mod.log.Errorf("bad pool label %q", req.Label)
+		resp.Status = int32(drpc.DaosInvalidInput)
+		return
+	}
+	if len(req.Svcreps) == 0 {
+		mod.log.Errorf("pool %q has zero svcreps", req.Uuid)
+		resp.Status = int32(drpc.DaosInvalidInput)
+		return
+	}
+	if _, err := mod.poolDB.FindPoolServiceByUUID(uuid); err == nil {
+		mod.log.Errorf("pool with uuid %q already exists", req.Uuid)
+		resp.Status = int32(drpc.DaosExists)
+		return
+	}
+	if _, err := mod.poolDB.FindPoolServiceByLabel(req.Label); err == nil {
+		mod.log.Errorf("pool with label %q already exists", req.Label)
+		resp.Status = int32(drpc.DaosExists)
+		return
+	}
+
+	ps := &system.PoolService{
```

Comment:
@Nasf-Fan: I think we can add it later, but we should probably also have the full set of known storage ranks as part of this information. The MS stores this information for later use during pool destroy/cleanup. We also can store some information about the per-rank tier storage allocations (scm/nvme), but we don't really do anything with it right now.